src/App.js

```diff
@@ -3,18 +3,84 @@ import Card from './Card';
 import { useState } from 'react';
 
 function App() {
-  const [checked, setChecked] = useState(false);
+  const [cardState, setCardState] = useState({
+    checked: false,
+    isEditable: false
+  });
+
+  const [cardText, setCardText] = useState({
+    headerText: 'Caption',
+    tempHeader: 'Caption',
+    bodyText: 'text...',
+    tempBody: 'text...'
+  });
+
   const changeStyleHandler = () => {
-    setChecked(!checked);
-  }
+    setCardState({
+      ...cardState,
+      checked: !cardState.checked,
+    });
+  };
+
+  const openEditModeHandler = () => {
+    setCardState({
+      checked: false,
+      isEditable: !cardState.isEditable
+    });
+  };
+
+  const headerTextChangeHandler = e => {
+    setCardText({
+      ...cardText,
+      headerText: e.target.value
+    });
+  };
+
+  const bodyTextChangeHandler = e => {
+    setCardText({
+      ...cardText,
+      bodyText: e.target.value
+    });
+  };
```

Comment:
These could be combined into one handler if pass the property name and set it dynamically.
Also, the concept is to update tempCard data rather that actual cardData. Since you should not update the data in storage unless you actually press a save icon.